### PR TITLE
Greeter-Spring added

### PR DESCRIPTION
**Pull request by TejasM**

### CODE REVIEW

**Comment by sgilda** [comment]
I tried to deploy this against 7.2.0.Alpha-SNAPSHOT and it hangs on the deploy at this point: 

$ mvn clean package jboss-as:deploy
[INFO] Scanning for projects...
[INFO]  
[INFO] ------------------------------------------------------------------------
[INFO] Building JBoss AS Quickstarts: Greeter Spring 7.1.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.4.1:clean (default-clean) @ jboss-as-greeter-spring ---
[INFO] Deleting /home/sgilda/GitRepos/quickstart-jdf/greeter-spring/target
[INFO] 
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ jboss-as-greeter-spring ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.3.1:compile (default-compile) @ jboss-as-greeter-spring ---
[INFO] Compiling 5 source files to /home/sgilda/GitRepos/quickstart-jdf/greeter-spring/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.5:testResources (default-testResources) @ jboss-as-greeter-spring ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/sgilda/GitRepos/quickstart-jdf/greeter-spring/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.3.1:testCompile (default-testCompile) @ jboss-as-greeter-spring ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.9:test (default-test) @ jboss-as-greeter-spring ---
[INFO] No tests to run.
[INFO] Surefire report directory: /home/sgilda/GitRepos/quickstart-jdf/greeter-spring/target/surefire-reports

---

##  T E S T S

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- maven-war-plugin:2.1.1:war (default-war) @ jboss-as-greeter-spring ---
[INFO] Packaging webapp
[INFO] Assembling webapp [jboss-as-greeter-spring] in [/home/sgilda/GitRepos/quickstart-jdf/greeter-spring/target/jboss-as-greeter-spring]
[INFO] Processing war project
[INFO] Copying webapp resources [/home/sgilda/GitRepos/quickstart-jdf/greeter-spring/src/main/webapp]
[INFO] Webapp assembled in [62 msecs]
[INFO] Building war: /home/sgilda/GitRepos/quickstart-jdf/greeter-spring/target/jboss-as-greeter-spring.war
[INFO] WEB-INF/web.xml already added, skipping
[INFO] 
[INFO] --- jboss-as-maven-plugin:7.1.1.Final:deploy (default-cli) @ jboss-as-greeter-spring ---
Apr 08, 2013 3:22:47 PM org.xnio.Xnio <clinit>
INFO: XNIO Version 3.0.3.GA
Apr 08, 2013 3:22:47 PM org.xnio.nio.NioXnio <clinit>
INFO: XNIO NIO Implementation Version 3.0.3.GA
Apr 08, 2013 3:22:47 PM org.jboss.remoting3.EndpointImpl <clinit>
INFO: JBoss Remoting version 3.2.3.GA

I will try against 7.1.1.next.

File: greeter-spring/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+4. This will deploy target/greeter-spring.war to the running instance of the server.
+
+If you don't have maven configured you can manually copy target/greeter-spring.war to JBOSS_HOME/standalone/deployments.
+
+Access the application
+----------------------
+
+The application will be running at the following URL: <http://localhost:8080/jboss-as-greeter-spring>
+
```
Comment (sgilda) [inline greeter-spring/README.md:74]: This is the wrong URL. It also should provided the link:

The application will be running at the following URL: http://localhost:8080/jboss-as-greeter-spring

**Comment by TejasM** [comment]
All the above fixed in the new commit, let me know if I should squash them. Sorry for all these small mistakes.

File: greeter-spring/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Source: <https://github.com/jboss-jdf/jboss-as-quickstart/>
+
+What is this?
+-------------
+
+The application this project produces is designed to be run on JBoss Enterprise Application Platform 6 or JBoss AS 7.
+
+When you deploy this example, two users are automatically created for you: emuster and jdoe. This data is located in the src/main/resources/init-db.sql file.
+
+Greeter-Spring differs from the standard greeter quickstart in that it uses Spring MVC for Mapping GET and POST request:
```
Comment (sgilda) [inline greeter-spring/README.md]: I would change this to the following. Note the tick marks (`) around the quickstart names to make them stand out:

The `greeter-spring` quickstart differes from the standard `greeter` quickstart in that it uses Spring MVC for mapping GET and POST request:

File: greeter-spring/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+* \<mvc:annotation-driven\> configured in src/main/webapp/WEB-INF/spring-mvc-context.xml tells spring to look for @RequestMapping in our controllers.
```
Comment (sgilda) [inline greeter-spring/README.md]: The `<mvc:annotation-driven\>` element configured in `src/main/webapp/WEB-INF/spring-mvc-context.xml` file tells Spring to look for `@RequestMapping` in our controllers.

We should probably always capitalize Spring.

**Comment by sgilda** [comment]
This looks much better! The explanation of the differences is great! Thanks for making the changes. :-)

The above comments are just suggestions. We should capitalize Spring and we usually put tick marks around file names, quickstart names, elements, etc., but if you don't want to make those changes, I can make them later.

Thanks again!

**Comment by sgilda** [comment]
One last thing: would you mind squashing the commits one more time? Thanks!

**Comment by TejasM** [comment]
@sgilda: Those are great suggestions I will put apply them both here and for rest of the PR's (and squash the commits), by today.

**Comment by sgilda** [comment]
Thanks for being so patient with me. :-)

**Comment by TejasM** [comment]
@sgilda: Updated and I should be the one thanking you, for doing my first major code/documentation review. Thank you!

File: greeter-spring/pom.xml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            <url>http://www.apache.org/licenses/LICENSE-2.0.html</url>
+        </license>
+    </licenses>
+
+    <properties>
+        <!-- Explicitly declaring the source encoding eliminates the following message: -->
+        <!-- [WARNING] Using platform encoding (UTF-8 actually) to copy filtered
+        resources, i.e. build is platform dependent! -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
```
Comment (sgilda) [inline greeter-spring/pom.xml:42]: Add the following (copy from the kitchensink/pom.xml for accurate formatting):

```
    <!-- Define the version of the JBoss BOMs we want to import to specify 
        tested stacks. -->
    <version.jboss.bom>1.0.4.Final</version.jboss.bom>
    <!-- Alternatively, comment out the above line, and un-comment the line 
        below to use version 1.0.4.Final-redhat-3 which is a release certified to 
        work with JBoss EAP 6. It requires you have access to the JBoss EAP 6 maven 
        repository. -->
    <!-- <version.jboss.bom>1.0.4.Final-redhat-3</version.jboss.bom> -->
```

File: greeter-spring/pom.xml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        <!-- Third Party dependencies -->
+        <version.standard.taglibs>1.1.2</version.standard.taglibs>
+        <version.commons.logging>1.1.1</version.commons.logging>
+
+        <!-- JBoss AS plugin for deployment -->
+        <version.jboss.as.maven.plugin>7.1.1.Final</version.jboss.as.maven.plugin>
+
+        <version.jboss.hibernate.bom>1.0.4.Final</version.jboss.hibernate.bom>
```
Comment (sgilda) [inline greeter-spring/pom.xml]: This can be removed. It's in the `<version.jboss.bom>`.

File: greeter-spring/pom.xml [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        <!-- [WARNING] Using platform encoding (UTF-8 actually) to copy filtered
+        resources, i.e. build is platform dependent! -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <!-- Third Party dependencies -->
+        <version.standard.taglibs>1.1.2</version.standard.taglibs>
+        <version.commons.logging>1.1.1</version.commons.logging>
+
+        <!-- JBoss AS plugin for deployment -->
+        <version.jboss.as.maven.plugin>7.1.1.Final</version.jboss.as.maven.plugin>
```
Comment (sgilda) [inline greeter-spring/pom.xml]: QS Tools reports: Version for [org.jboss.as.plugins:jboss-as-maven-plugin:${version.jboss.as.maven.plugin}] isn't using the recommended property name: version.jboss.maven.plugin

The property names were updated a while ago to make them simpler and more standard.

version.jboss.as.maven.plugin was changed to version.jboss.maven.plugin

File: greeter-spring/pom.xml [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+                <artifactId>jboss-javaee-6.0-with-hibernate</artifactId>
+                <version>${version.jboss.hibernate.bom}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+
+            <dependency>
+                <groupId>org.jboss.bom</groupId>
+                <artifactId>jboss-javaee-6.0-with-spring</artifactId>
+                <version>${version.jboss.spring.bom}</version>
```
Comment (sgilda) [inline greeter-spring/pom.xml]: <version>${version.jboss.bom}</version>

File: greeter-spring/pom.xml [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                of artifacts. We use this here so that we always get the correct versions
+                of artifacts. Here we use the jboss-javaee-web-6.0 stack (you can read this
+                as the JBoss stack of the Java EE Web Profile 6 APIs), and we use version
+                2.0.0.Beta1 which is the latest release of the stack. You can actually use
+                this stack with any version of JBoss AS that implements Java EE 6, not just
+                JBoss AS 7! -->
+            <dependency>
+                <groupId>org.jboss.bom</groupId>
+                <artifactId>jboss-javaee-6.0-with-hibernate</artifactId>
+                <version>${version.jboss.hibernate.bom}</version>
```
Comment (sgilda) [inline greeter-spring/pom.xml]: <version>${version.jboss.bom}</version>

**Comment by sgilda** [comment]
@TejasM : I ran the QS Tools utility against this quickstart and it uncovered a few issues with the pom.xml file. I made the comments at the appropriate line numbers.

We're almost there!

You may want to run the report against all your new Spring quickstarts. There were some updates to the tool that won't be available until sometime tomorrow, so you may want to wait a day or so to catch all the problems. :-)

**Comment by TejasM** [comment]
@sgilda: Okay sure I will take a look tomorrow before updating. 

Regarding ${version.jboss.bom}, if I use the same property for the -with-spring bom then I would have to update <version.jboss.bom> to 1.0.5.CR3, because spring bom was added only add at 1.0.5.CR3. Is this alright?

**Comment by sgilda** [comment]
@TejasM : I will check with @rafabene. For now, I would leave it as it is.

**Comment by rafabene** [comment]
Hi @TejasM . Yes. You right. both BOM should use 1.0.5.CR4 (released today) using the same property name ${version.jboss.bom}.

Thank you! :)

**Comment by TejasM** [comment]
@rafabene: Ah okay, thanks!

**Comment by sgilda** [comment]
@TejasM : @rafabene said the updated QS Tool is on Maven Central now.

**Comment by TejasM** [comment]
Okay awesome, will update after checking

**Comment by TejasM** [comment]
@sgilda: Updated, there is a warning about using 1.0.5.CR4 as the bom version but I think that is alright given @rafabene's comment.

**Comment by sgilda** [comment]
I agree. I'll review once more. :-)

File: greeter-spring/pom.xml [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            <plugin>
+                <groupId>org.jboss.as.plugins</groupId>
+                <artifactId>jboss-as-maven-plugin</artifactId>
+                <version>${version.jboss.as.maven.plugin}</version>
+            </plugin>
+            <!-- Surefire plugin before 2.9 version is buggy -->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.9</version>
+            </plugin>
```
Comment (sgilda) [inline greeter-spring/pom.xml:196]: <version>${version.surefire.plugin}</version>

File: greeter-spring/pom.xml [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.3.1</version>
+                <configuration>
+                    <source>1.6</source>
+                    <target>1.6</target>
+                </configuration>
+            </plugin>
+            <!-- Deployent on AS from console -->
```
Comment (sgilda) [inline greeter-spring/pom.xml:186]: The above should be:
`
            <plugin>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${version.compiler.plugin}</version>
                <configuration>
                    <source>${maven.compiler.source}</source>
                    <target>${maven.compiler.target}</target>
                </configuration>
            </plugin>
`

**Comment by sgilda** [comment]
@TejasM : You missed a few hard-coded versions in the dependencies. I marked them in the file. :-)

**Comment by TejasM** [comment]
@sgilda: Updated, I will update other PR's later today.

**Comment by sgilda** [comment]
To run the new version of the plugin, you need the version in the command:

```
    mvn org.jboss.maven.plugins:maven-qstools-plugin:1.0.0.CR3:check
```

**Comment by sgilda** [comment]
Merged!

**Pull request closed without merging**